test/e2e/atlas/helper_test.go

```diff
@@ -302,9 +313,43 @@ func deleteClusterForProject(projectID, clusterName string) error {
 	}
 	watchCmd := exec.Command(cliPath, watchArgs...)
 	watchCmd.Env = os.Environ()
-	// this command will fail with 404 once the cluster is deleted
-	// we just need to wait for this to close the project
-	_ = watchCmd.Run()
+	return watchCmd.Run()
+}
+
+func removeTerminationProtectionFromCluster(projectID, clusterName string) error {
+	cliPath, err := e2e.AtlasCLIBin()
+	if err != nil {
+		return err
+	}
+	args := []string{
+		clustersEntity,
+		"update",
+		clusterName,
+		"--disableTerminationProtection",
+	}
+	if projectID != "" {
+		args = append(args, "--projectId", projectID)
+	}
+	updateCmd := exec.Command(cliPath, args...)
+	updateCmd.Env = os.Environ()
+	if resp, err := updateCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("error updating cluster %w: %s", err, string(resp))
+	}
+
+	return watchCluster(projectID, clusterName)
+}
+
+func deleteClusterForProject(projectID, clusterName string) error {
+	if err := internalDeleteClusterForProject(projectID, clusterName); err != nil {
+		if !strings.Contains(err.Error(), "CANNOT_TERMINATE_CLUSTER_WHEN_TERMINATION_PROTECTION_ENABLED") {
```

Comment:
is this err generated immediately or only after the watchCluster timeout?
I wonder if it's not better to do a getCluster, check the `terminationProtectionEnabled` flag, then removeTerminationProtectionFromCluster (if this is enabled) and then internalDeleteClusterForProject

Reply:
immediately, watch happens only if the resource passes validation.

I personally think getCluster is worse, only a subset of clusters are not being deleted because of termination protection if we add getCluster we will impact every single one of the clusters as opposed to only the small subset, leading to a much slower flow.

Reply:
if it's immediate I'm ok with this flow, thanks for checking